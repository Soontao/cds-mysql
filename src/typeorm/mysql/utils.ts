
/**
 * equal string without case sensitive
 * 
 * @param s1 first string
 * @param s2 second string
 * @returns true if both two string are equal without case sensitive
 */
export function equalWithoutCase(s1: string, s2: string) {
  if (s1 === undefined && s2 === undefined) {
    return true;
  }
  if (s1?.length === 0 && s2?.length === 0) {
    return true;
  }
  if(s1 === s2) {
    return true;
  }
  if (s1?.length === s2?.length && s1?.toLowerCase() === s2?.toLowerCase()) {
    return true;
  }

  return false;
}