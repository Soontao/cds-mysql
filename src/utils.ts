
/**
 * overwrite cds compiler type mapping
 */
export function overwriteCDSCoreTypes() {

  const { cdsToSqlTypes } = require("@sap/cds-compiler/lib/render/utils/common");

  cdsToSqlTypes.sqlite = {
    "cds.Binary": "CHAR",
    "cds.hana.BINARY": "CHAR",
    "cds.hana.SMALLDECIMAL": "DECIMAL",
  };

}