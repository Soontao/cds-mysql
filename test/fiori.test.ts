import { cwdRequireCDS, setupTest } from "cds-internal-tool";
import { doAfterAll } from "./utils";



describe("fiori draft Test Suite", () => {

  const cds = cwdRequireCDS();
  const client = setupTest(__dirname, "./resources/fiori");
  client.defaults.auth = { username: "alice", password: "admin" };


  beforeAll(() => { jest.spyOn(cds.db, "run"); });

  afterAll(doAfterAll);


  it("should support get metadata", async () => {
    const response = await client.get("/fiori/$metadata");
    expect(response.status).toBe(200);
    expect(response.data).toMatch(/Persons/);
  });

  const ID = "95f96069-e831-4e30-9567-37a1490b9385";

  it("should support to create instance", async () => {
    const response = await client.post("/fiori/Persons", {
      ID,
      Name: "Theo Sun",
      Age: 26,
    });
    expect(response.data?.error).toBeUndefined();
    expect(response.status).toBe(201);
    expect(response.data).toMatchSnapshot();
  });

});