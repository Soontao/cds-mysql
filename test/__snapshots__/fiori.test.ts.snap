// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fiori draft Test Suite should support to create instance 1`] = `
Object {
  "@odata.context": "$metadata#Persons/$entity",
  "Address": null,
  "Age": 25,
  "HasActiveEntity": false,
  "HasDraftEntity": false,
  "ID": "95f96069-e831-4e30-9567-37a1490b9385",
  "IsActiveEntity": false,
  "Name": "Theo Sun",
}
`;
